processor/src/test/java/org/mapstruct/ap/test/builder/factory/BuilderFactoryMapper.java

```diff
@@ -0,0 +1,40 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.builder.factory;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.ObjectFactory;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Filip Hrisafov
+ */
+@Mapper
+public abstract class BuilderFactoryMapper {
+
+    public static final BuilderFactoryMapper INSTANCE = Mappers.getMapper( BuilderFactoryMapper.class );
+
+    public abstract Person map(PersonDto source);
+
+    @ObjectFactory
```

Comment:
Perhaps good to test whether the mechanism without `ObjectFactory` also works.

Reply:
Done